traffic_ops/testing/api/v4/servicecategories_test.go

```diff
@@ -233,3 +269,139 @@ func DeleteTestServiceCategories(t *testing.T) {
 		}
 	}
 }
+
+func DeleteTestServiceCategoriesInvalidName(t *testing.T) {
+
+	delResp, reqInf, err := TOSession.DeleteServiceCategory("invalid", client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected no serviceCategory with that key found but got: %v - alerts: %+v", err, delResp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusNotFound {
+		t.Fatalf("Expected 404 status code, got %v", reqInf.StatusCode)
+	}
+}
+
+func GetTestServiceCategoriesByInvalidName(t *testing.T) {
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", "invalid")
+	resp, _, _ := TOSession.GetServiceCategories(opts)
+	if len(resp.Response) > 0 {
+		t.Errorf("Expected 0 response, but got many %v", resp)
+	}
+}
+
+func VerifyPaginationSupportServiceCategories(t *testing.T) {
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("orderby", "id")
+	resp, _, err := TOSession.GetServiceCategories(opts)
+	if err != nil {
+		t.Fatalf("cannot get Service Categories: %v - alerts: %+v", err, resp.Alerts)
+	}
+	serviceCategories := resp.Response
+	if len(serviceCategories) < 2 {
+		t.Fatalf("Need at least 2 Service Categories in Traffic Ops to test pagination support, found: %d", len(serviceCategories))
+	}
+
+	opts.QueryParameters = url.Values{}
+	opts.QueryParameters.Set("orderby", "id")
+	opts.QueryParameters.Set("limit", "1")
+	serviceCategoriesWithLimit, _, err := TOSession.GetServiceCategories(opts)
+	if err == nil {
+		if !reflect.DeepEqual(serviceCategories[:1], serviceCategoriesWithLimit.Response) {
+			t.Error("expected GET Service Categories with limit = 1 to return first result")
+		}
+	} else {
+		t.Error("Error in getting Service Categories by limit")
```

Comment:
This should print what the error actually was

Reply:
Added err in error print statement